src/main/java/net/snowflake/client/core/SFStatement.java

```diff
@@ -194,8 +197,10 @@ SFBaseResultSet executeQueryInternal(
     }
 
     Object result =
-        executeHelper(
-            sql, StmtUtil.SF_MEDIA_TYPE, parameterBindings, describeOnly, internal, asyncExec);
+        queryId.isPresent()
```

Comment:
If user want to get Result for a specific query, call
`StmtUtil.getQueryResultJSON(queryId.get(), session)`